### PR TITLE
Remove expired polls from overview page

### PR DESCRIPTION
**Pull request by personthingman2**
This is my first contribution to an open source project so please tell me if I'm doing anything wrong.
This is a fix for #1168. I moved the logic into `GetGlobalStatus` because `MemorizeMessage` is only run when a message is received, meaning users would have to restart their client for an expired poll to disappear. As a result `msPoll` now contains the entire message XML so it can be processed in `GetGlobalStatus`.
The "Poll: " and "Foundation Poll: " concats were removed because it looked silly to me.

### CODE REVIEW

File: src/main.cpp
Comment (tomasbrod) [inline src/main.cpp]: Avoid converting things to string if possible.

File: src/main.cpp
Comment (tomasbrod) [inline src/main.cpp]: Unsafe strtoll. Please add error and exception handling or use another function (ToString maybe?).

File: src/main.cpp
Comment (tomasbrod) [inline src/main.cpp]: I do not like this.

**Review by tomasbrod — CHANGES_REQUESTED** [review]
Please put your logic in a function and wrap it in a try-catch so it does not crash wallet on invalid data.
Otherwise good. Thank you for contributing.

**Comment by personthingman2** [comment]
I've moved the logic into the new function`GetPoll()`, added `uint_64_t uPollExpiration` which removes the need to convert to string, and put a try/catch around the remaining `stoll()`. I don't believe an additional try/catch is required around the function call in `GetGlobalStatus()` because that entire function is already in a try/catch.

File: src/main.cpp
Comment (personthingman2) [reply src/main.cpp]: Could you elaborate? I don't see what's wrong with this.

**Review by tomasbrod — APPROVED** [review]

**Review by tomasbrod — CHANGES_REQUESTED** [review]
I accidentally set status to approve.
About that no polls message: if it is a separate line, sure, show that message so it does not appear empty. But if it is under status field, no polls message would just takes up space.

**Comment by personthingman2** [comment]
I've changed `GetPoll()` to `GetCurrentOverviewTabPoll()` and added a try/catch around the function call with a log error.

The "No current polls" message is only shown in the "Current Poll: " field of the overview tab so it does not clutter any other field.

Here is a screenshot of what it looks like: 
screenshot from 2018-08-11 15-57-32

**Comment by iFoggz** [comment]
label poll is not used for anything but poll

**Review by iFoggz — APPROVED** [review]
tACK 569bc8d; apon another approval I think due to this being a cosmetic fix more then anything it'll be a small include for the upcoming release milestone if @denravonska and @tomasbrod concur.

File: src/main.cpp
Comment (iFoggz) [inline src/main.cpp]: Translations should be considered here. 
`GlobalStatusStruct.poll = _("Error obtaining last poll");`

File: src/main.cpp
Comment (iFoggz) [inline src/main.cpp]: Translations:
poll = _("No current polls");

**Review by iFoggz — CHANGES_REQUESTED** [review]
Actually thinking about this. We should consider use the _() and do translations where we can to increase the readability across languages where we can.

File: src/main.cpp
Comment (iFoggz) [reply src/main.cpp]: I agree if anything a log entry about the exception should be put. Remove translation and log the e.what() under debug

File: src/main.cpp
Comment (iFoggz) [reply src/main.cpp]: Short but sweet.

**Comment by personthingman2** [comment]
I will wait on adding translations until #1262 is merged.

Do we want it to say "No current polls" or "None"? I think the former sounds better in the context of the overview page.

**Comment by iFoggz** [comment]
i would say don't worry about the translations part and it can be done at a later point when we do another translation update before release :) having the _() wont break your code. no worries there -- so just add the requested _() in a commit and make sure all reviews are satisfied and we can move to approval and merge. i'll update the translation strings at next release

you wouldn't be manually adding to the bitcoinstrings.cpp file. there's a script for that and would be in a separate pr for updated information

**Comment by personthingman2** [comment]
Is this PR appropriate in its current state or would you guys like any additional changes?

**Review by tomasbrod — APPROVED** [review]

**Review by iFoggz — APPROVED** [review]

**Comment by cyrossignol** [comment]
Any reason to worry about the full contract message appearing in the output of `getmininginfo` now for command-line users?

```json
"MiningInfo 2": "<MT>poll</MT><MK>What_do_you_call_a_positively_charged_seal?\n</MK><MV><TITLE>What_do_you_call_a_positively_charged_seal?\n</TITLE><DAYS>14</DAYS><QUESTION>What_do_you_call_a_positively_charged_seal?\n</QUESTION><ANSWERS>A_Sealion;God_dammit_Foxifi!</ANSW ERS><SHARETYPE>3</SHARETYPE><URL>https://i.imgur.com/1D4iQZl.jpg</URL><EXPIRATION>1544712997</EXPIRATION></MV><MA>A</MA><MPK></MPK><MS>MEQCIHwhNJRKus9MwSYWo6epSyfaNytv2/IFFp5fCVUDNZwUAiBUS2U68h4y+ptkQ7tnUr/YGO8bL1Gz4EbIrE1207g2zA==</MS>",
```

**Comment by philipswift** [comment]
With jokes like that...yes! ;)

**Comment by cyrossignol** [comment]
@philipswift What did the network admin seal say when the IP address stopped working?

<details>
ARP! ARP! ARP!
</details>

**Pull request merged**
